### PR TITLE
Reorganize project structure and implement app launcher drawer portlet

### PR DESCRIPTION
**Pull request by SaraBoutej**

### CODE REVIEW

File: app-center-webapps/webpack.common.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    context: path.resolve(__dirname, '.'),
+    // set the entry point of the application
+    // can use multiple entry
+
+    entry: {
+        adminSetup: "./src/main/webapp/vue-apps/adminSetup/adminSetup.js",
+        userSetup: "./src/main/webapp/vue-apps/userSetup/userSetup.js",
+        myApplications: "./src/main/webapp/vue-apps/myApplications/myApplications.js",
+        appLauncher: "./src/main/webapp/vue-apps/appLauncher/main.js"
```
Comment (azayati) [inline app-center-webapps/webpack.common.js]:
```suggestion
        appLauncher: "./src/main/webapp/vue-apps/appLauncher/appLauncher.js"
```

File: app-center-webapps/src/main/webapp/vue-apps/appLauncher/main.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import launcherApp from './components/appLauncher.vue';
```
Comment (azayati) [inline app-center-webapps/src/main/webapp/vue-apps/appLauncher/main.js]:
```suggestion
import AppLauncherApp from './components/AppLauncher.vue';
```

File: app-center-webapps/src/main/webapp/vue-apps/appLauncher/components/appLauncher.vue [hunk 3]
```diff
@@ -0,0 +1,58 @@
+<template>
+  <v-app
+    id="app-center-launcher"
```
Comment (azayati) [inline app-center-webapps/src/main/webapp/vue-apps/appLauncher/components/appLauncher.vue:3]: Remove id if it is not used

File: app-center-webapps/src/main/webapp/WEB-INF/portlet.xml [hunk 4]
```diff
@@ -74,4 +74,26 @@
     <portlet-preferences>
     </portlet-preferences>
   </portlet>
+  <!-- ================================================================ -->
+  <!-- = App Launcher My Applications Portlet = -->
```
Comment (azayati) [inline app-center-webapps/src/main/webapp/WEB-INF/portlet.xml]:
```suggestion
  <!-- = App Center App Launcher Portlet = -->
```

File: app-center-webapps/src/main/webapp/WEB-INF/portlet.xml [hunk 5]
```diff
@@ -74,4 +74,26 @@
...
+    <init-param>
+      <name>portlet-view-dispatched-file-path</name>
+      <value>/html/administration.html</value>
+    </init-param>
+    <supports>
+      <mime-type>text/html</mime-type>
+    </supports>
+    <supported-locale>fr</supported-locale>
+    <portlet-info>
+      <title>App center my applications launcher portlet</title>
```
Comment (azayati) [inline app-center-webapps/src/main/webapp/WEB-INF/portlet.xml]:
```suggestion
      <title>App center app launcher portlet</title>
```

File: app-center-webapps/src/main/webapp/WEB-INF/portlet.xml [hunk 6]
```diff
@@ -74,4 +74,26 @@
...
+      <keywords>App center Launcher portlets</keywords>
```
Comment (azayati) [inline app-center-webapps/src/main/webapp/WEB-INF/portlet.xml]:
```suggestion
      <keywords>App center portlets</keywords>
```

File: app-center-webapps/src/main/webapp/html/administration.html [hunk 7]
```diff
@@ -0,0 +1,9 @@
+<div class="VuetifyApp">
+    <div id="app-center-launcher">
```
Comment (azayati) [inline app-center-webapps/src/main/webapp/html/administration.html]:
```suggestion
    <div id="appLauncher">
```

File: app-center-webapps/src/main/webapp/html/administration.html [hunk 8]
```diff
@@ -0,0 +1,9 @@
...
+        <script>
+            require(['SHARED/appCenterLauncherBundle'], function(appLauncher) {
```
Comment (azayati) [inline app-center-webapps/src/main/webapp/html/administration.html]:
```suggestion
            require(['SHARED/appLauncherBundle'], function(appLauncherApp) {
```

File: app-center-webapps/src/main/webapp/html/administration.html [hunk 9]
```diff
@@ -0,0 +1,9 @@
...
+                appLauncher.init();
```
Comment (azayati) [inline app-center-webapps/src/main/webapp/html/administration.html]:
```suggestion
                appLauncherApp.init();
```

File: app-center-webapps/src/main/webapp/vue-apps/appLauncher/main.js [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+
+const url = `/app-center/vueLocales/locale_${lang}.json`;
+
+export function init() {
+//getting locale ressources
+    exoi18n.loadLanguageAsync(lang, url)
+        .then(i18n => {
+            // init Vue app when locale ressources are ready
+            new Vue({
+                render: h => h(launcherApp),
```
Comment (azayati) [inline app-center-webapps/src/main/webapp/vue-apps/appLauncher/main.js]:
```suggestion
                render: h => h(AppLauncherApp),
```

File: app-center-webapps/src/main/webapp/vue-apps/appLauncher/main.js [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+                i18n,
+                vuetify,
+            }).$mount('#app-center-launcher');
```
Comment (azayati) [inline app-center-webapps/src/main/webapp/vue-apps/appLauncher/main.js]:
```suggestion
            }).$mount('#appLauncher');
```

File: app-center-webapps/src/main/webapp/WEB-INF/portlet.xml [hunk 12]
```diff
@@ -74,4 +74,26 @@
     <portlet-preferences>
     </portlet-preferences>
   </portlet>
+  <!-- ================================================================ -->
+  <!-- = App Launcher My Applications Portlet = -->
+  <!-- ================================================================== -->
+  <portlet>
+    <portlet-name>appCenterLauncher</portlet-name>
```
Comment (azayati) [inline app-center-webapps/src/main/webapp/WEB-INF/portlet.xml]:
```suggestion
    <portlet-name>AppCenterAppLauncherPortlet</portlet-name>
```

File: app-center-webapps/src/main/webapp/WEB-INF/portlet.xml [hunk 13]
```diff
@@ -74,4 +74,26 @@
...
+    <display-name xml:lang="EN">App center my applications portlet</display-name>
```
Comment (azayati) [inline app-center-webapps/src/main/webapp/WEB-INF/portlet.xml]:
```suggestion
    <display-name xml:lang="EN">App center app launcher portlet</display-name>
```

File: app-center-webapps/src/main/webapp/WEB-INF/portlet.xml [hunk 14]
```diff
@@ -74,4 +74,26 @@
...
+    <portlet-class>org.exoplatform.commons.api.portlet.GenericDispatchedViewPortlet</portlet-class>
+    <init-param>
+      <name>portlet-view-dispatched-file-path</name>
+      <value>/html/administration.html</value>
```
Comment (azayati) [inline app-center-webapps/src/main/webapp/WEB-INF/portlet.xml]:
```suggestion
      <value>/html/appLauncher.html</value>
```

File: app-center-webapps/src/main/webapp/WEB-INF/gatein-resources.xml [hunk 15]
```diff
@@ -43,6 +43,15 @@
     </module>
   </portlet>
 
+  <portlet>
+    <name>appCenterLauncher</name>
```
Comment (azayati) [inline app-center-webapps/src/main/webapp/WEB-INF/gatein-resources.xml]:
```suggestion
    <name>AppCenterAppLauncherPortlet</name>
```

File: app-center-webapps/src/main/webapp/WEB-INF/gatein-resources.xml [hunk 16]
```diff
@@ -43,6 +43,15 @@
...
+    <module>
+      <depends>
+        <module>appCenterLauncherBundle</module>
```
Comment (azayati) [inline app-center-webapps/src/main/webapp/WEB-INF/gatein-resources.xml]:
```suggestion
        <module>appLauncherBundle</module>
```

File: app-center-webapps/src/main/webapp/WEB-INF/gatein-resources.xml [hunk 17]
```diff
@@ -91,4 +100,22 @@
       <module>eXoVueI18n</module>
     </depends>
   </module>
+
+  <module>
+    <name>appCenterLauncherBundle</name>
```
Comment (azayati) [inline app-center-webapps/src/main/webapp/WEB-INF/gatein-resources.xml]:
```suggestion
    <name>appLauncherBundle</name>
```

File: app-center-webapps/src/main/webapp/WEB-INF/conf/portal/portal/sharedlayout.xml [hunk 18]
```diff
@@ -0,0 +1,132 @@
...
+                        <portlet-ref>NotificationPopoverToolbarPortlet</portlet-ref>
+                    </portlet>
+                    <access-permissions>*:/platform/users</access-permissions>
+                    <show-info-bar>false</show-info-bar>
+                </portlet-application>
+            </container>
+            <container id="ApplicationLauncherPortlet" template="system:/groovy/portal/webui/container/UIContainer.gtmpl">
+                <portlet-application>
+                    <portlet>
+                        <application-ref>platformNavigation</application-ref>
```
Comment (azayati) [inline app-center-webapps/src/main/webapp/WEB-INF/conf/portal/portal/sharedlayout.xml]:
```suggestion
                        <application-ref>app-center</application-ref>
```

File: app-center-webapps/src/main/webapp/WEB-INF/conf/portal/portal/sharedlayout.xml [hunk 19]
```diff
@@ -0,0 +1,132 @@
...
+                        <portlet-ref>appCenterLauncher</portlet-ref>
```
Comment (azayati) [inline app-center-webapps/src/main/webapp/WEB-INF/conf/portal/portal/sharedlayout.xml]:
```suggestion
                        <portlet-ref>AppCenterAppLauncherPortlet</portlet-ref>
```

File: app-center-webapps/src/main/webapp/WEB-INF/conf/portal/portal/sharedlayout.xml [hunk 20]
```diff
@@ -0,0 +1,132 @@
...
+                <portlet-application>
+                    <portlet>
+                        <application-ref>platformNavigation</application-ref>
+                        <portlet-ref>NotificationPopoverToolbarPortlet</portlet-ref>
+                    </portlet>
+                    <access-permissions>*:/platform/users</access-permissions>
+                    <show-info-bar>false</show-info-bar>
+                </portlet-application>
+            </container>
+            <container id="ApplicationLauncherPortlet" template="system:/groovy/portal/webui/container/UIContainer.gtmpl">
```
Comment (azayati) [inline app-center-webapps/src/main/webapp/WEB-INF/conf/portal/portal/sharedlayout.xml]:
```suggestion
            <container id="AppCenterAppLauncherPortlet" template="system:/groovy/portal/webui/container/UIContainer.gtmpl">
```

**Review by azayati — CHANGES_REQUESTED** [review]
- See review
- .vue files should be with capital first letter

File: app-center-webapps/webpack.common.js (hunk 1 above)
Comment (SaraBoutej) [reply app-center-webapps/webpack.common.js]: Done!

File: app-center-webapps/src/main/webapp/vue-apps/appLauncher/main.js (hunk 2 above)
Comment (SaraBoutej) [reply app-center-webapps/src/main/webapp/vue-apps/appLauncher/main.js]: Done!

File: app-center-webapps/src/main/webapp/vue-apps/appLauncher/components/appLauncher.vue (hunk 3 above)
Comment (SaraBoutej) [reply app-center-webapps/src/main/webapp/vue-apps/appLauncher/components/appLauncher.vue:3]: used!

File: app-center-webapps/src/main/webapp/WEB-INF/portlet.xml (hunk 4 above)
Comment (SaraBoutej) [reply app-center-webapps/src/main/webapp/WEB-INF/portlet.xml]: Done

File: app-center-webapps/src/main/webapp/WEB-INF/portlet.xml (hunk 5 above)
Comment (SaraBoutej) [reply app-center-webapps/src/main/webapp/WEB-INF/portlet.xml]: Done!

File: app-center-webapps/src/main/webapp/WEB-INF/portlet.xml (hunk 6 above)
Comment (SaraBoutej) [reply app-center-webapps/src/main/webapp/WEB-INF/portlet.xml]: Done!

File: app-center-webapps/src/main/webapp/html/administration.html (hunk 7 above)
Comment (SaraBoutej) [reply app-center-webapps/src/main/webapp/html/administration.html]: Done!

File: app-center-webapps/src/main/webapp/html/administration.html (hunk 8 above)
Comment (SaraBoutej) [reply app-center-webapps/src/main/webapp/html/administration.html]: Done!

File: app-center-webapps/src/main/webapp/html/administration.html (hunk 9 above)
Comment (SaraBoutej) [reply app-center-webapps/src/main/webapp/html/administration.html]: Done!

File: app-center-webapps/src/main/webapp/vue-apps/appLauncher/main.js (hunk 10 above)
Comment (SaraBoutej) [reply app-center-webapps/src/main/webapp/vue-apps/appLauncher/main.js]: Done!

File: app-center-webapps/src/main/webapp/vue-apps/appLauncher/main.js (hunk 11 above)
Comment (SaraBoutej) [reply app-center-webapps/src/main/webapp/vue-apps/appLauncher/main.js]: Done!

File: app-center-webapps/src/main/webapp/WEB-INF/portlet.xml (hunk 12 above)
Comment (SaraBoutej) [reply app-center-webapps/src/main/webapp/WEB-INF/portlet.xml]: Done!

File: app-center-webapps/src/main/webapp/WEB-INF/portlet.xml (hunk 13 above)
Comment (SaraBoutej) [reply app-center-webapps/src/main/webapp/WEB-INF/portlet.xml]: Done!

File: app-center-webapps/src/main/webapp/WEB-INF/portlet.xml (hunk 14 above)
Comment (SaraBoutej) [reply app-center-webapps/src/main/webapp/WEB-INF/portlet.xml]: Done!

File: app-center-webapps/src/main/webapp/WEB-INF/gatein-resources.xml (hunk 15 above)
Comment (SaraBoutej) [reply app-center-webapps/src/main/webapp/WEB-INF/gatein-resources.xml]: Done!

File: app-center-webapps/src/main/webapp/WEB-INF/gatein-resources.xml (hunk 16 above)
Comment (SaraBoutej) [reply app-center-webapps/src/main/webapp/WEB-INF/gatein-resources.xml]: Done!

File: app-center-webapps/src/main/webapp/WEB-INF/gatein-resources.xml (hunk 17 above)
Comment (SaraBoutej) [reply app-center-webapps/src/main/webapp/WEB-INF/gatein-resources.xml]: Done!

File: app-center-webapps/src/main/webapp/WEB-INF/conf/portal/portal/sharedlayout.xml (hunk 18 above)
Comment (SaraBoutej) [reply app-center-webapps/src/main/webapp/WEB-INF/conf/portal/portal/sharedlayout.xml]: Done!

File: app-center-webapps/src/main/webapp/WEB-INF/conf/portal/portal/sharedlayout.xml (hunk 19 above)
Comment (SaraBoutej) [reply app-center-webapps/src/main/webapp/WEB-INF/conf/portal/portal/sharedlayout.xml]: Done!

File: app-center-webapps/src/main/webapp/WEB-INF/conf/portal/portal/sharedlayout.xml (hunk 20 above)
Comment (SaraBoutej) [reply app-center-webapps/src/main/webapp/WEB-INF/conf/portal/portal/sharedlayout.xml]: Done!

**Review by azayati — APPROVED** [review]
LGTM

**Pull request merged**